src/plugins/dashboard/public/dashboard_app/top_nav/dashboard_editing_toolbar.tsx

```diff
@@ -79,32 +87,70 @@ export function DashboardEditingToolbar() {
     [stateTransferService, search.session, trackUiMetric]
   );
 
-  const getVisTypeQuickButton = (visTypeName: string) => {
-    const visType =
-      getVisualization(visTypeName) || getVisTypeAliases().find(({ name }) => name === visTypeName);
+  const createNewEmbeddable = useCallback(
+    async (embeddableFactory: EmbeddableFactory) => {
+      if (trackUiMetric) {
+        trackUiMetric(METRIC_TYPE.CLICK, embeddableFactory.type);
+      }
+
+      const explicitInput = await embeddableFactory.getExplicitInput();
+      const newEmbeddable = await dashboardContainer.addNewEmbeddable(
+        embeddableFactory.type,
+        explicitInput
+      );
+
+      if (newEmbeddable) {
+        pluginServices.getServices().notifications.toasts.addSuccess({
```

Comment:
Nit: we generally like to destructure required services at the top of the file when possible. Additionally, we may not need this toast here.

Reply:
Fixed for now. 
Didn't remove https://github.com/elastic/kibana/pull/146421#discussion_r1050714694 and this code is still used for viz lib